### PR TITLE
Fix spec tests on windows, enable spec tests for windows in CI

### PR DESCRIPTION
**Pull request by loganek**

### CODE REVIEW

File: .github/workflows/compilation_on_windows.yml [hunk 1]
```diff
@@ -39,6 +39,12 @@ on:
   # allow to be triggered manually
   workflow_dispatch:
 
+env:
+  # For Spec Test
+  DEFAULT_TEST_OPTIONS: "-s spec -b"
```
Comment (wenyongh) [inline .github/workflows/compilation_on_windows.yml:44]: Can we use `-P` option to run the spec test parallelly?

**Review by wenyongh — APPROVED** [review]
LGTM

File: .github/workflows/compilation_on_windows.yml (hunk 1 above)
Comment (loganek) [reply .github/workflows/compilation_on_windows.yml:44]: There's something in Python script that won't work when enabling parallel execution. I'm yet to investigate it but I thought we can enable those tests in a single-threaded mode.

Comment (wenyongh) [reply .github/workflows/compilation_on_windows.yml:44]: Got it, thanks.

**Pull request merged**